courses/fundamentals_of_ada/070_subprograms.rst

```diff
@@ -77,68 +76,58 @@ A Little "Preaching" About Names
    function Square_Root (V: Real) return Real;
    function Is_Open (V: Valve) return Boolean;
 
-========================
-Declarations and Bodies
-========================
-
-----------
-Examples
-----------
-
-.. include:: examples/070_subprograms/declarations_and_bodies.rst
+========
+Syntax
+========
 
-:url:`https://learn.adacore.com/training_examples/fundamentals_of_ada/070_subprograms.html#declarations-and-bodies`
+-------------------------
+ Specification and Body
+-------------------------
 
---------------------------
-Subprogram Declarations
---------------------------
+* Subprogram specification is the external (user) **interface**
 
-* Define the external (user) interface
+   - **Declaration** and **specification** are used synonymously
 
-   - Provide the subprogram specification
-   - *Declaration* and *specification* are often used synonymously
+* Specification may be required in
 
-* Required in some circumstances
+   - Package specification
+   - Recursion
+   - (Optional otherwise)
 
-   - You'll see them...
+* Subprogram body is the **implementation**
 
-* Optional in other circumstances
+-------------------------------------------
+Procedure Specification Syntax (Simplified)
+-------------------------------------------
 
 .. code:: Ada
 
    procedure Swap (A, B : in out Integer);
-   function F (X : Real) return Real;
 
--------------------
-Subprogram Bodies
--------------------
+.. code:: Ada
 
-* Provide the implementation
-* Define execution behavior
+   procedure_specification ::=
+      procedure identifier
+        ( parameter_specification
+        { ; parameter_specification} )
 
-   - Procedure body
+   parameter_specification ::=
+      identifier_list : mode subtype_mark [ := expression ]
 
-      .. code:: Ada
+   mode ::= [in] | out | in out
 
-         procedure Swap (A, B : in out Integer) is
-           Temp : Integer := A;
-         begin
-           A := B;
-           B := Temp;
-         end Swap;
+------------------------------------------
+Function Specification Syntax (Simplified)
+------------------------------------------
 
-   - Function body
+.. code:: Ada
 
-      .. code:: Ada
+   function F (X : Real) return Real;
 
-         function F (X : Real) return Real is
-         begin
-           return X + 3.0 * X;
-         end F;
+* Same as :ada:`procedure`
```

Comment:
You say "same as" and then you switch names around.
e.g procedure name is a "defining_name" but function name can be "operator" or "identifier"
same with "parameter_profile" and "formal_part"